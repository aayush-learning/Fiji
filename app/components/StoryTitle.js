import React, { PureComponent } from 'react'

import { StyleSheet, TouchableOpacity, View, Text, FlatList, Image, Dimensions } from 'react-native'
import PropTypes from 'prop-types'

export default class StoryTitle extends PureComponent {
  state = Dimensions.get("window")
  handler = dims => this.setState(dims)

  componentWillMount() {
    Dimensions.addEventListener("change", this.handler);
  }

  componentWillUnmount() {
    Dimensions.removeEventListener("change", this.handler); // This is an important to stop updating state after unmount
  }

  _onPress = () => {
    this.props.onPressItem(this.props.title)
  }

  render() {
    const {width, height} = this.state
    //const mode = height > width ? "portrait" : "landscape";
    return (
      <TouchableOpacity onPress={ this._onPress } style={[styles.TouchableStyle, { width:Dimensions.get('window').width * 0.3, marginLeft:Dimensions.get('window').width * 0.02 }]}>
        <View style={styles.TextViewStyle}>
          <Text style={styles.TextStyle}>
            {this.props.title.title}
          </Text>
        </View>
        <View style={styles.ImageViewStyle}>
          <Image style={styles.ImageStyle}/>
        </View> 
      </TouchableOpacity>
    )
  }
}

StoryTitle.propTypes = {
  title: PropTypes.object,
  onPressItem: PropTypes.func
}

const styles = StyleSheet.create({
  ImageViewStyle:{ 
    flex:3, 
    alignItems:'center',
    backgroundColor:'red',
    padding:'4%',
    borderBottomLeftRadius:30,
    borderBottomRightRadius:30
  },
  ImageStyle:{
    borderColor:'black',
    borderWidth:2
  },
  TextViewStyle:{
    flex:2,
    borderColor:'black',
    borderWidth:2,
    flexDirection:'row',
    flexWrap:'wrap',
    //justifyContent:'flex-start',
    alignItems:'center',
    //justifyContent:'center',
    alignContent:'center',
    backgroundColor:'#19a4f2',
    borderTopLeftRadius:30,
    borderTopRightRadius:30,
  },
  TextStyle:{
    fontWeight:'bold', 
    fontSize:20, 
    color:'black',
    marginTop:'2%',
  },
  TouchableStyle:{
    flex:1,
    height:140, 
    //width:Dimensions.get('window').width * 0.3, 
    //marginRight:'3%',
    //marginLeft:Dimensions.get('window').width * 0.02,
    marginTop:'20%',
    marginBottom:'12%',
    borderTopLeftRadius:30,
    borderTopRightRadius:30,
    borderBottomLeftRadius:30,
    borderBottomRightRadius:30
  }
});