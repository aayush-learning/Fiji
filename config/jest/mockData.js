import { fetchUsersFailure } from '../../app/redux/users';

const contentJson = require('../seed/content.json')
const usersJson = require('../seed/users.json')
const userAliceJson = require('../seed/user_alice.json')
const storyJson = require('../seed/story.json')

export const titles = contentJson['docs'].filter(function (doc) {
  return doc._id.startsWith('storytitle:')
})

export const stories = storyJson['docs'].filter(function (doc) {
  return doc._id.startsWith('story:')
})

export const games = contentJson['docs'].filter(function (doc) {
  return doc._id.startsWith('game:')
})

export const gameThemeData = contentJson['docs'].filter(function (doc) {
  return doc._id.startsWith('gametheme:')
})

export const stickersData = contentJson['docs'].filter(function (doc) {
  return doc._id.startsWith('sticker:caterpillar:')
})

export const stickerPacksData = contentJson['docs'].filter(function (doc) {
  return doc._id.startsWith('sticker-pack:')
})

export const highScoresData = contentJson['docs'].filter(function (doc) {
  return doc._id.startsWith('high-score:')
})

export const story_alice_in_wonderland = storyJson['docs'].filter(function (doc) {
  return doc._id == 'story:alice-in-wonderland'
})[0]

export const users = usersJson['docs']

export const messages = userAliceJson['docs'].filter(function(doc) {
  return doc._id.startsWith('chat:mad-hatter:')
})