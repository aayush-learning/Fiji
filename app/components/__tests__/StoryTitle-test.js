import 'react-native';
import React from 'react';
import StoryTitle from '../StoryTitle';
import { titles } from '../../../config/jest/mockData'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <StoryTitle 
      title = { titles[0] }
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
