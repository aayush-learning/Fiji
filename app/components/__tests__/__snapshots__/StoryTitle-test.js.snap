// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderBottomLeftRadius": 30,
      "borderBottomRightRadius": 30,
      "borderColor": "black",
      "borderTopLeftRadius": 30,
      "borderTopRightRadius": 30,
      "borderWidth": 2,
      "flex": 1,
      "height": 140,
      "marginBottom": "12%",
      "marginHorizontal": "5%",
      "marginTop": "20%",
      "opacity": 1,
      "width": 105,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "backgroundColor": "#19a4f2",
        "borderTopLeftRadius": 30,
        "borderTopRightRadius": 30,
        "flex": 2,
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "flex-start",
        "padding": "1%",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontSize": 20,
          "fontWeight": "bold",
          "marginTop": "1%",
        }
      }
    >
      Alice in Wonderland
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 3,
        "padding": "4%",
      }
    }
  >
    <Image
      style={
        Object {
          "borderColor": "black",
          "borderRadius": 60,
          "borderWidth": 2,
          "height": "100%",
          "width": "100%",
        }
      }
    />
  </View>
</View>
`;
